services/frontend-service/src/ui/components/button/ExpandButton.tsx

```diff
@@ -0,0 +1,93 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+import { useCallback, useState } from 'react';
+import * as React from 'react';
+import { Button } from './button';
+import { PlainDialog } from '../dialog/ConfirmationDialog';
+
+/**
+ * Two buttons combined into one.
+ * Inspired by GitHubs merge button.
+ * Displays one normal button on the left, and one arrow on the right to select a different option.
+ */
+export type ExpandButtonProps = {
+    onClickSubmit: (shouldLockToo: boolean) => void;
```

Comment:
I would have imagined a more generic solution but this actually works as well.

Reply:
yeah, currently it's very specifically build to this one use case of "deploy / deploy&lock".
E.g. we also only support one button that "pops up". But we can extend this once it's needed.